### DISCUSSION TITLE
Issue validating Azure subsription in-flight

### DISCUSSION THREAD

**Discussion by donk-msft** [Azure/PSRule.Rules.Azure#2150] [Q&A] [answered]
I'm new to PSRule and though I'm following the documentation, I'm running into issues/questions.

I've installed the required modules:

- Install-Module -Name PowerShellGet -Repository PSGallery -Scope CurrentUser -Force
- Install-Module -Name PSRule -Repository PSGallery -Scope CurrentUser -AllowPrerelease
- Install-Module -Name PSRule.Rules.Azure -Repository PSGallery -Scope CurrentUser -AllowPrerelease

When I run 'Export-AzRuleData -Subscription <subId> from PowerShell, I get a <guid>.json file.

When I subsequently run 'Invoke-PSRule -InputPath 'c:\temp\PSRule' -Module 'PSRule.Rules.Azure' -As Summary', nothing happens, just new line.

_Note: the required syntax for -InputPath switch is unclear to me and 'get-help Invoke-PSRule -detailed' doesn't provide additional information._

When I run ''Invoke-PSRule -Module 'PSRule.Rules.Azure' -As Summary' and provide the full path+filename as inputObject, I get a warning: WARNING: Target object '3c46b38c47d7697e1a83b4af6ff9db26ae02b0ce' has not been processed because no matching rules were found.

I also created a default 'ps-rule.yaml' file in my working directory, but no apparent effect.

What am I doing wrong?

**Comment by donk-msft** [comment, marked as answer]
Update:
When I run 'Invoke-PSRule -InputPath '.' -Module 'PSRule.Rules.Azure', it runs the assessment.

**Reply by BernieWhite** [reply]
@GitHubUser1339 Great that your worked it out.

Something like `Invoke-PSRule -InputPath 'c:\temp\PSRule\*.json' -Module 'PSRule.Rules.Azure' -As Summary` or `Invoke-PSRule -InputPath 'c:\temp\PSRule\' -Module 'PSRule.Rules.Azure' -As Summary` should also work if your exports are in the `c:\temp\PSRule\` directory.

Piping to `Invoke-PSRule` or using `-InputObject` PSRule is going to expect either JSON or a `System.IO.FileInfo` object instead of a file name.

So something like `Get-ChildItem '*.json' | Invoke-PSRule -Module 'PSRule.Rules.Azure' -As Summary` should work.
